src/settings/components/ExternalDataSourcesConfig/ExternalDataSourcesSettings/ExternalDataSourcesSettings.js

```diff
@@ -0,0 +1,92 @@
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import arrayMutators from 'final-form-arrays';
+
+import ExternalDataSourcesView from '../ExternalDataSourcesView/ExternalDataSourcesView';
+import ExternalDataSourcesLookup from '../ExternalDataSourcesLookup/ExternalDataSourcesLookup';
+import ExternalDataSourcesFormModal from '../ExternaldataSourcesFormModal/ExternalDataSourcesFormModal';
+import ExternalDataSourceForm from '../ExternalDataSourceForm/ExternalDataSourceForm';
+
+const EDITING = 'edit';
+const CREATING = 'create';
+const VIEWING = 'view';
+
+const ExternalDataSourcesSettings = ({
+  onDelete,
+  onCancel,
+  onSave,
+  externalKbs,
+  onSubmit,
+  initialValues
+}) => {
+  const [mode, setMode] = useState(VIEWING);
+  const [externalDataSource, setExternalDataSource] = useState();
+
+  return (
+    <>
+      <ExternalDataSourcesLookup
+        externalKbs={externalKbs}
+        onClick={() => setMode(CREATING)}
+        onSelectedExternalDataSource={(_e, eds) => {
+          setExternalDataSource(eds);
+        }}
+      />
+      {
+        externalDataSource &&
+          <ExternalDataSourcesView
+            externalDataSourceId={externalDataSource.id}
+            externalKbs={externalKbs}
+            initialValues={initialValues}
+            onCancel={onCancel}
+            onClick={() => setMode(EDITING)}
+            onClose={() => setExternalDataSource()}
+            onDelete={onDelete}
+            onSave={onSave}
+            onSubmit={onSubmit}
+          />
+      }
+      <ExternalDataSourcesFormModal
+        initialValues={mode === CREATING ?
+          {
+            active: false,
+            activationEnabled: false,
+            rectype: 1,
+            supportsHarvesting: true,
+            type: 'org.olf.kb.adapters.GOKbOAIAdapter',
+          }
+          :
+          { ...externalDataSource }}
```

Comment:
I believe this means that after an edit a subsequent "edit" modal won't have the changes. Might be best to split the ExtenralDataSourcesFormModal into two, and move one over to ExternalDataSourcesLookup and one to ExternalDataSourcesView.